### PR TITLE
[TACHYON-821] Replace System.setProperty with TachyonConf.set in inte…

### PR DESCRIPTION
**Pull request by miasantreble**
…gration-tests

https://tachyon.atlassian.net/browse/TACHYON-821

### CODE REVIEW

File: integration-tests/src/test/java/tachyon/client/RemoteBlockInStreamIntegrationTest.java [hunk 1]
```diff
@@ -78,19 +78,18 @@ public RemoteBlockInStreamIntegrationTest(String[] classes) {
...
   @Before
   public final void before() throws Exception {
     mLocalTachyonCluster = new LocalTachyonCluster(10000, 1000, Constants.GB);
-    System.setProperty(Constants.WORKER_DATA_SERVER, mDataServerClass);
-    System.setProperty(Constants.USER_REMOTE_BLOCK_READER, mRemoteReaderClass);
     mLocalTachyonCluster.start();
     mLocalTachyonCluster.getWorkerTachyonConf().set(Constants.USER_REMOTE_READ_BUFFER_SIZE_BYTE,
         "100");
     mMasterTachyonConf = mLocalTachyonCluster.getMasterTachyonConf();
+    TachyonConf workerTachyonConf = mLocalTachyonCluster.getWorkerTachyonConf();
```
Comment (apc999) [inline integration-tests/src/test/java/tachyon/client/RemoteBlockInStreamIntegrationTest.java]: `mLocalTachyonCluster.getWorkerTachyonConf()` twice (line 87 and 90)

Comment (miasantreble) [inline integration-tests/src/test/java/tachyon/client/RemoteBlockInStreamIntegrationTest.java]: done

**Comment by gpang** [comment]
I have a question: Is it ok for these properties to be set after "mLocalTachyonCluster.start()", and why does that work?

**Comment by miasantreble** [comment]
@gpang I think the idea here is that LocalTachyonCluster.mWorkerConf is a static object, so the order of "start() -> setConf() -> getConf()" should work. you can see RemoteBlockInStreamIntegrationTest#before() was doing this even before this PR

**Comment by gpang** [comment]
Thanks for the info. I can see how setting the conf before creating a client works, but does setting the conf after the master and worker start actually affect the master and worker?

File: integration-tests/src/test/java/tachyon/client/RemoteBlockInStreamIntegrationTest.java [hunk 2]
```diff
@@ -78,19 +78,17 @@ public RemoteBlockInStreamIntegrationTest(String[] classes) {
...
     mLocalTachyonCluster = new LocalTachyonCluster(10000, 1000, Constants.GB);
-    System.setProperty(Constants.WORKER_DATA_SERVER, mDataServerClass);
-    System.setProperty(Constants.USER_REMOTE_BLOCK_READER, mRemoteReaderClass);
     mLocalTachyonCluster.start();
-    mLocalTachyonCluster.getWorkerTachyonConf().set(Constants.USER_REMOTE_READ_BUFFER_SIZE_BYTE,
-        "100");
     mMasterTachyonConf = mLocalTachyonCluster.getMasterTachyonConf();
+    TachyonConf workerTachyonConf = mLocalTachyonCluster.getWorkerTachyonConf();
+    workerTachyonConf.set(Constants.USER_REMOTE_READ_BUFFER_SIZE_BYTE, "100");
+    workerTachyonConf.set(Constants.WORKER_DATA_SERVER, mDataServerClass);
```
Comment (uronce-cc) [inline integration-tests/src/test/java/tachyon/client/RemoteBlockInStreamIntegrationTest.java]: In [`mLocalTachyonCluster.start()`](https://github.com/amplab/tachyon/pull/1352/files#diff-3468a5e5623a435ee4a657759d2b191bL90), the worker is started, if `WORKER_DATA_SERVER` is specified here, how can a worker decide which data server to use?

**Comment by miasantreble** [comment]
@gpang @uronce-cc thanks for the comments, I've fixed the issue by first creating a TachyonConf object, set all the attributes needed, then pass the conf object to start method. Let me know if there are further issues

**Comment by apc999** [comment]
I think the current approach of this PR works. 
+1 LGTM

Meanwhile, we shall refactor `LocalTachyonCluster` to make it cleaner.

**Comment by uronce-cc** [comment]
+1 LGTM

**Pull request merged**